src/plugins/inputs/ctcp.ts

```diff
@@ -0,0 +1,37 @@
+import Msg, {MessageType} from "../../models/msg";
+import {PluginInputHandler} from "./index";
+
+const commands = ["ctcp"];
+
+const input: PluginInputHandler = function ({irc}, chan, cmd, args) {
+	if (args.length < 2) {
+		chan.pushMessage(
+			this,
+			new Msg({
+				type: MessageType.ERROR,
+				text: "Usage: /ctcp <nick> <ctcp_type>",
+			})
+		);
+		return;
+	}
+
+	chan.pushMessage(
+		this,
+		new Msg({
+			type: MessageType.CTCP_REQUEST,
+			ctcpMessage: `"${args.slice(1).join(" ")}" to ${args[0]}`,
```

Comment:
you could move `target` above and use it here, which removes the slice too.

also todo below